internal/api/middleware.go

```diff
@@ -14,21 +15,42 @@ func recoverMiddleware(logger logr.Logger) func(http.Handler) http.Handler {
 	return func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			defer func() {
-				if r := recover(); r != nil {
-					var msg string
-					switch x := r.(type) {
-					case string, error:
-						msg = fmt.Sprintf("panic: %s", x)
+				if rec := recover(); rec != nil {
+					// Don't recover if the request is aborted, otherwise the
```

Comment:
Nitpick: the "otherwise" in the comment is confusing, but I think the meaning is: `Don't recover if the request is aborted because an aborted request doesn't report an error.`?